packages/syntax-core/src/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { type ReactElement } from "react";
+import Checkbox from "../Checkbox/Checkbox";
+import Box from "../Box/Box";
+
+type OptionProps = {
+  value: string;
+  label: string;
+};
+
+export type checkedOptionProps = Record<string, boolean>;
+
+/**
+ * [CheckboxGroup](https://cambly-syntax.vercel.app/?path=/docs/components-typography--docs) is a component that renders a list of checkboxes.
+ */
+const CheckboxGroup = ({
+  options,
+  onFormChange,
+  selections,
+  size = "md",
+}: {
+  /**
+   * The list of labels and values for each checkbox
+   */
+  options: OptionProps[];
+  /**
+   * The callback to be called when a checkbox value changes
+   */
+  onFormChange: (optionIds: checkedOptionProps) => void;
+  /**
+   * The list of values that are selected
+   */
+  selections: Record<string, boolean>;
+  /**
+   * The size of the checkboxes and icons
+   *
+   * * `sm`: 16px
+   * * `md`: 24px
+   *
+   * @defaultValue "md"
+   */
+  size?: "sm" | "md";
+}): ReactElement => {
+  const onChange = (
+    { target: { checked } }: React.ChangeEvent<HTMLInputElement>,
+    id: string,
+  ) => {
+    const updatedSelections = { ...selections, [id]: checked };
+    onFormChange(updatedSelections);
+  };
+
+  return (
+    <Box display="flex" direction="column" gap={3}>
```

Comment:
@zprong ideally folks would be able to use `CheckboxGroup` as follows:

```tsx
<CheckboxGroup>
  <Checkbox label="soccer">Soccer</Checkbox>
  <Checkbox label="baseball">Baseball</Checkbox>
  <Checkbox label="basketball">Basketball</Checkbox>
</CheckboxGroup>
```

That makes it more composable (everything is still a React element) and avoid use having to pass in big objects to components.

We can make `CheckboxGroup` dumb and not support other props so it would just be:
```tsx
<Box display="flex" direction="column" gap={3}>
  {children}
</Box>
```

If you do want it to be smarter, I would recommend using React context to pass the state around from `CheckboxGroup` to `Checkbox`.

A good example to look at:
* https://react-spectrum.adobe.com/react-spectrum/CheckboxGroup.html
* https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/checkbox/src/CheckboxGroup.tsx